src/main/java/de/bsi/secvisogram/csaf_cms_backend/service/AdvisoryService.java

```diff
@@ -205,20 +222,143 @@ public String changeAdvisoryWorkflowState(String advisoryId, String revision, Wo
         }
         AdvisoryWrapper existingAdvisoryNode = AdvisoryWrapper.createFromCouchDb(existingAdvisoryStream);
 
-        AuditTrailWrapper auditTrail = AuditTrailWorkflowWrapper.createNewFrom(newWorkflowState, existingAdvisoryNode.getWorkflowState())
+        AuditTrailWrapper auditTrail = AdvisoryAuditTrailWorkflowWrapper.createNewFrom(newWorkflowState, existingAdvisoryNode.getWorkflowState())
+                .setDocVersion(existingAdvisoryNode.getDocumentTrackingVersion())
+                .setOldDocVersion(existingAdvisoryNode.getDocumentTrackingVersion())
                 .setAdvisoryId(advisoryId)
                 .setCreatedAtToNow()
                 .setChangeType(ChangeType.Update)
-                .setUser("Mustermann")
-                .setDocVersion(existingAdvisoryNode.getDocumentTrackingVersion())
-                .setOldDocVersion(existingAdvisoryNode.getDocumentTrackingVersion());
+                .setUser("Mustermann");
         this.couchDbService.writeDocument(UUID.randomUUID(), auditTrail.auditTrailAsString());
 
         existingAdvisoryNode.setWorkflowState(newWorkflowState);
         existingAdvisoryNode.setRevision(revision);
         return this.couchDbService.updateDocument(existingAdvisoryNode.advisoryAsString());
     }
 
+    /**
+     * Adds a comment to the advisory
+     *
+     * @param advisoryId  the ID of the advisory to add the comment to
+     * @param commentJson the comment to add as JSON string, requires a commentText
+     * @return a tuple of ID and revision of the added comment
+     * @throws DatabaseException when there are database errors
+     * @throws IOException       when there are errors in JSON handling
+     */
+    public IdAndRevision addComment(String advisoryId, String commentJson) throws DatabaseException, IOException {
+
+        UUID commentId = UUID.randomUUID();
+
+        CommentWrapper newComment = CommentWrapper.createNewFromJson(advisoryId, commentJson);
+
+        AuditTrailWrapper auditTrail = CommentAuditTrailWrapper.createNew(newComment)
+                .setCommentId(commentId.toString())
+                .setChangeType(ChangeType.Create);
+
+        String user = newComment.getOwner();
+        if (user != null) {
+            auditTrail.setUser(user);
+        }
+
+        String commentRevision = couchDbService.writeDocument(commentId, newComment.commentAsString());
+        couchDbService.writeDocument(UUID.randomUUID(), auditTrail.auditTrailAsString());
+
+        InputStream advisoryStream = couchDbService.readDocumentAsStream(newComment.getAdvisoryId());
+        AdvisoryWrapper advisory = AdvisoryWrapper.createFromCouchDb(advisoryStream);
+        String field = newComment.getField();
+        advisory.addCommentId(Objects.requireNonNullElse(field, ""), commentId.toString());
+        couchDbService.updateDocument(advisory.advisoryAsString());
```

Comment:
When adding a new comment, it will also be added to the advisory JSON at a `$comment` node. This is where it happens and currently this PR is aligned with this behavior (in tests etc.)

A downside is that it triggers an update to the advisory (and new revision) every time a comment is added. Another option would be to outsource this modification of the advisory JSON to the client.

@domachine do you have any opinion on this?

Reply:
As discussed we'll change the behavior to let the client generate IDs for each object node in the advisory and reference those in a `nodeId` field of the comment. More documentation on that will follow.